### PR TITLE
docs: RBE, Docker sandbox and bazel fixes

### PR DESCRIPTION
**Pull request by lizan**
Description:
Added RBE and Docker sandbox how-to.
Make macOS ASAN working again.

Risk Level:
Testing:
Docs Changes: Developer docs
Release Notes:

### CODE REVIEW

File: bazel/README.md [hunk 1]
```diff
@@ -84,6 +84,40 @@ Bazel can also be built with the Docker image used for CI, by installing Docker
...
+
+```
+bazel build //source/exe:envoy-static --config=remote-clang \
+    --remote_cache=grpcs://remotebuildexecution.googleapis.com \
+    --remote_executor=grpcs://remotebuildexecution.googleapis.com \
+    --remote_instance_name=projects/envoy-ci/instances/default_instance
+```
+
+Change the value of `--remote_cache`, `--remote_executor` and `--remote_instance_name` for your remote build services. Tests can
+be run in remote execution too.
```
Comment (jmarantz) [inline bazel/README.md]: I don't see what this paragraph adds to the above example. The above example seems complete.

File: bazel/README.md [hunk 2]
```diff
@@ -84,6 +84,40 @@ Bazel can also be built with the Docker image used for CI, by installing Docker
...
+
+Note: Currently the test run configuration in `.bazelrc` doesn't download test binaries and test logs,
```
Comment (jmarantz) [inline bazel/README.md]: downloading failed test logs by default seems really useful, particularly if something flakes. If you can't get the log then you may be out of luck. 

However, downloading the test binaries seems less useful and more expensive. Is there a way to tease these options apart?

File: bazel/README.md [hunk 3]
```diff
@@ -84,6 +84,40 @@ Bazel can also be built with the Docker image used for CI, by installing Docker
...
+part of the CI is running with the hosted [GCP RBE](https://blog.bazel.build/2018/10/05/remote-build-execution.html) service.
+
+To build Envoy with a remote build services, run Bazel with your remote build service flags and with `--config=remote-clang`.
+For example the following command runs build with the GCP RBE service used in CI:
+
+```
+bazel build //source/exe:envoy-static --config=remote-clang \
+    --remote_cache=grpcs://remotebuildexecution.googleapis.com \
+    --remote_executor=grpcs://remotebuildexecution.googleapis.com \
+    --remote_instance_name=projects/envoy-ci/instances/default_instance
```
Comment (jmarantz) [inline bazel/README.md]: anyone can use these googleapis for free?

Comment (lizan) [reply bazel/README.md]: No, that's why I the paragraph:

> Change the value of `--remote_cache`, `--remote_executor` and `--remote_instance_name` for your remote build services. Tests can be run in remote execution too.

File: bazel/README.md (hunk 2 above)
Comment (lizan) [reply bazel/README.md]: That is what I'm asking in https://github.com/bazelbuild/bazel/issues/8934

**Comment by lizan** [comment]
@jmarantz gently ping.

File: bazel/README.md (hunk 2 above)
Comment (jmarantz) [reply bazel/README.md]: Cool. You might want to reference that bug in a comment here?

Comment (jmarantz) [reply bazel/README.md]: actually that issue is closed now; can you update the default to download test logs?

Comment (lizan) [reply bazel/README.md]: The fix is not in released bazel yet.

**Pull request merged**